app/components/edit/EditServices.jsx

```diff
@@ -32,21 +31,16 @@ class EditServices extends Component {
    * @returns {void}
    */
   handleServiceChange(key, service) {
-    const { services } = this.state;
-    services[key] = service;
-    this.setState({
-      services,
-    }, () => {
-      this.props.handleServiceChange(this.state);
-    });
+    const { handleServiceChange } = this.props;
+    handleServiceChange({ services: { [key]: service } });
```

Comment:
I think this might be changing behavior. We were previously pulling all the services off state, updating the one we changed, and then passing all the services back into `handleServiceChange`. Here we are creating an empty services object with just the service that is changing.

I personally prefer the new approach of just passing the service that is changing (I might even argue we can just pass the key & service as parameters instead of wrapping them in a 'services' object). But, I think we need to update the `handleServiceChange` (from props) that is handling this to make sure it isn't overwriting the services that exist at that parent component level and instead just merging this changed one in with its existing services.

Reply:
Good call. Let me revert this one change back, and I can take care of refactoring the handlers in a different PR.